src/material/select/select.ts

```diff
@@ -611,7 +614,7 @@ export abstract class _MatSelectBase<C>
   ngOnChanges(changes: SimpleChanges) {
     // Updating the disabled state is handled by `mixinDisabled`, but we need to additionally let
     // the parent form field know to run change detection when the disabled state changes.
-    if (changes['disabled']) {
+    if (changes['disabled'] || changes['userAriaDescribedBy']) {
```

Comment:
Why is this change necessary? Since the `aria-describedby` is on the select it shouldn't be required.

Reply:
Without it, this test fails:

```
it('should support user binding to `aria-describedby`', fakeAsync(() => {
  fixture.componentInstance.ariaDescribedBy = 'test';
  fixture.detectChanges();
  expect(select.getAttribute('aria-describedby')).toBe('test');
}));
```

As a result of this PR (and #24292 for MatChipList and #19587 for MatInput), the `aria-describedby` is no longer set in `@Component({ host })`. Rather, it is set using `nativeElement.setAttribute` in `MatSelect#setDescribedByIds`, which is called by `MatFormField#_syncDescribedByIds`, which in turn is called when `stateChanges` updates.

So, `userAriaDescribedBy` has to trigger a `stateChanges.next()` in order for the `aria-describedby` to actually be set.

Reply:
...and for that matter, MatChipList needs unit tests for this too! I've submitted #24657. In that one I used a setter for `userAriaDescribedBy` to call `stateChanges.next()`. Let me know if you'd prefer that here too.